query-node/mappings/src/content/app.ts

```diff
@@ -0,0 +1,123 @@
+import { MemberId } from '@joystream/types/primitives'
+import { App } from 'query-node/dist/model'
+import { logger } from '../common'
+
+export async function processCreateAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: ICreateApp
+): Promise<void> {
+  const { name, appMetadata } = message
+  const appId = await getAppId(store, event)
+
+  const isAppExists = await store.get(App, {
+    where: {
+      name: name,
+    },
+  })
+
+  if (isAppExists) {
+    logger.error('App already exists', { name })
+    return
+  }
+
+  const newApp = new App({
+    name,
+    id: appId,
+    createdById: memberId.toString(),
+    websiteUrl: appMetadata?.websiteUrl || undefined,
+    useUri: appMetadata?.useUri || undefined,
+    smallIcon: appMetadata?.smallIcon || undefined,
+    mediumIcon: appMetadata?.mediumIcon || undefined,
+    bigIcon: appMetadata?.bigIcon || undefined,
+    oneLiner: appMetadata?.oneLiner || undefined,
+    description: appMetadata?.description || undefined,
+    termsOfService: appMetadata?.termsOfService || undefined,
+    platforms: appMetadata?.platforms || undefined,
+    category: appMetadata?.category || undefined,
+    authKey: appMetadata?.authKey || undefined,
+  })
+  await store.save<CreateApp>(newApp)
+  logger.info('App has been created', { name })
+}
+
+export async function processUpdateAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IUpdateApp
+): Promise<void> {
+  const { appId, appMetadata } = message
+
+  const app = await getAppByIdAndMemberId(store, appId, memberId)
+
+  if (!app) {
+    logger.error("App doesn't exists or doesn't belong to the member", { appId, memberId: memberId.toString() })
+    return
+  }
+
+  app.websiteUrl = appMetadata?.websiteUrl || app.websiteUrl
+  app.useUri = appMetadata?.useUri || app.useUri
+  app.smallIcon = appMetadata?.smallIcon || app.smallIcon
+  app.mediumIcon = appMetadata?.mediumIcon || app.mediumIcon
+  app.bigIcon = appMetadata?.bigIcon || app.bigIcon
+  app.oneLiner = appMetadata?.oneLiner || app.oneLiner
+  app.description = appMetadata?.description || app.description
+  app.termsOfService = appMetadata?.termsOfService || app.termsOfService
+  app.platforms = appMetadata?.platforms || app.platforms
+  app.category = appMetadata?.category || app.category
+  app.authKey = appMetadata?.authKey || app.authKey
+
+  await store.save<App>(app)
+  logger.info('App has been updated', { appId })
+}
+
+export async function processDeleteAppMessage(
+  store: DatabaseManager,
+  event: SubstrateEvent,
+  memberId: MemberId,
+  message: IDeleteApp
+): Promise<void> {
+  const { appId } = message
+
+  const app = await getAppByIdAndMemberId(store, appId, memberId)
+
+  if (!app) {
+    logger.error("App doesn't exists or doesn't belong to the member", { appId, memberId: memberId.toString() })
+    return
+  }
+
+  await store.remove<App>(new App({ id: appId }))
```

Comment:
You can just pass the `app` object returned in line 86, no need to create new one.
```suggestion
  await store.remove<App>(app)
```